modules/sdk-coin-avaxp/src/lib/exportTxBuilder.ts

```diff
@@ -0,0 +1,98 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { NotSupported, TransactionType } from '@bitgo/sdk-core';
+import { AtomicTransactionBuilder } from './atomicTransactionBuilder';
+import {
+  BaseTx,
+  ExportTx,
+  PlatformVMConstants,
+  SECPTransferOutput,
+  TransferableOutput,
+  Tx,
+  UnsignedTx,
+} from 'avalanche/dist/apis/platformvm';
+import { BN } from 'avalanche';
+import { AmountOutput } from 'avalanche/dist/apis/evm/outputs';
+
+export class ExportTxBuilder extends AtomicTransactionBuilder {
+  private _amount: BN;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.Export;
+  }
+
+  /**
+   *
+   * @param value
+   */
+  amount(value: BN | string): this {
+    const valueBN = BN.isBN(value) ? value : new BN(value);
+    this.validateAmount(valueBN);
+    this._amount = valueBN;
+    return this;
+  }
+
+  initBuilder(tx: Tx): this {
+    super.initBuilder(tx);
+    const baseTx: BaseTx = tx.getUnsignedTx().getTransaction();
+    if (!this.verifyTxType(baseTx)) {
+      throw new NotSupported('Transaction cannot be parsed or has an unsupported transaction type');
+    }
+    this._externalChainId = baseTx.getDestinationChain();
+    this._amount = (baseTx.getExportOutputs()[0].getOutput() as any as AmountOutput).getAmount();
+    return this;
+  }
+
+  static get txType(): number {
+    return PlatformVMConstants.EXPORTTX;
+  }
+
+  verifyTxType(baseTx: BaseTx): baseTx is ExportTx {
+    return baseTx.getTypeID() === ExportTxBuilder.txType;
+  }
+
+  protected get outsMethod(): string {
+    return 'getExportOutputs';
+  }
+
+  /**
+   *
+   * @protected
+   */
+  protected buildAvaxpTransaction(): void {
+    const { inputs, outputs, credentials } = this.createInputOutput(this._amount.add(this.transaction._fee));
+    this.transaction.setTransaction(
+      new Tx(
+        new UnsignedTx(
```

Comment:
if you serialize a signed tx, then de-serialize it, it seems like you'll get an unsigned tx back.

is that intentional?

Reply:
You are right. Now, I avoid that case
https://github.com/BitGo/BitGoJS/blob/7df888eca68bbc60505d837cdf8729c69f4d387e/modules/sdk-coin-avaxp/src/lib/exportTxBuilder.ts#L78-L79